buffer/src/main/java/io/netty/buffer/AbstractByteBuf.java

```diff
@@ -1349,7 +1349,7 @@ public boolean equals(Object o) {
 
     @Override
     public int compareTo(ByteBuf that) {
-        return this == that? 0 : ByteBufUtil.compare(this, that);
+        return this == that ? 0 : ByteBufUtil.compare(this, that);
```

Comment:
What about just do this check in `ByteBUfUtil.compare` ?

Reply:
@normanmaurer , yes it's OK to move this check to `ByteBufUtil.compare`, if so, then the `ByteBufUtil.equal` should also add this check to make the logic consistence. IMO.

Reply:
> the ByteBufUtil.equal should also add this check to make the logic consistence. IMO.

Yes makes sense